Description: migrate eigen2 to eigen3
Author: Anton Gladky <gladk@debian.org> 
Bug-Debian: http://bugs.debian.org/726644
Last-Update: 2013-10-17

Index: kstars/CMakeLists.txt
===================================================================
--- kstars.orig/CMakeLists.txt	2013-11-09 17:45:22.758523776 +0100
+++ kstars/CMakeLists.txt	2013-11-09 17:45:22.754523934 +0100
@@ -15,8 +15,8 @@
 # some configure checks needed for kstars
 include(CheckIncludeFiles)
 
-find_package(Eigen2 2.0.3)
-macro_log_feature(EIGEN2_FOUND "Eigen2" "Required to build KStars." "http://eigen.tuxfamily.org" TRUE "2.0.3" "")
+find_package(Eigen3 REQUIRED)
+macro_log_feature(EIGEN3_FOUND "Eigen3" "Required to build KStars." "http://eigen.tuxfamily.org" TRUE "2.0.3" "")
 
 macro_optional_find_package(CFitsio)
 macro_bool_to_01(CFITSIO_FOUND HAVE_CFITSIO_H)
@@ -46,13 +46,14 @@
 enable_testing(true)
 
 
-if (EIGEN2_FOUND)
+if (EIGEN3_FOUND)
    include_directories(${CMAKE_CURRENT_BINARY_DIR})
+   include_directories(${EIGEN3_INCLUDE_DIR})
    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/config-kstars.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-kstars.h )
 
    add_subdirectory( doc )
    add_subdirectory( kstars )
-endif (EIGEN2_FOUND)
+endif (EIGEN3_FOUND)
 
 
 add_subdirectory(datahandlers)
Index: kstars/cmake/modules/FindEigen3.cmake
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ kstars/cmake/modules/FindEigen3.cmake	2013-11-09 17:45:22.754523934 +0100
@@ -0,0 +1,81 @@
+# - Try to find Eigen3 lib
+#
+# This module supports requiring a minimum version, e.g. you can do
+#   find_package(Eigen3 3.1.2)
+# to require version 3.1.2 or newer of Eigen3.
+#
+# Once done this will define
+#
+#  EIGEN3_FOUND - system has eigen lib with correct version
+#  EIGEN3_INCLUDE_DIR - the eigen include directory
+#  EIGEN3_VERSION - eigen version
+
+# Copyright (c) 2006, 2007 Montel Laurent, <montel@kde.org>
+# Copyright (c) 2008, 2009 Gael Guennebaud, <g.gael@free.fr>
+# Copyright (c) 2009 Benoit Jacob <jacob.benoit.1@gmail.com>
+# Redistribution and use is allowed according to the terms of the 2-clause BSD license.
+
+if(NOT Eigen3_FIND_VERSION)
+  if(NOT Eigen3_FIND_VERSION_MAJOR)
+    set(Eigen3_FIND_VERSION_MAJOR 2)
+  endif(NOT Eigen3_FIND_VERSION_MAJOR)
+  if(NOT Eigen3_FIND_VERSION_MINOR)
+    set(Eigen3_FIND_VERSION_MINOR 91)
+  endif(NOT Eigen3_FIND_VERSION_MINOR)
+  if(NOT Eigen3_FIND_VERSION_PATCH)
+    set(Eigen3_FIND_VERSION_PATCH 0)
+  endif(NOT Eigen3_FIND_VERSION_PATCH)
+
+  set(Eigen3_FIND_VERSION "${Eigen3_FIND_VERSION_MAJOR}.${Eigen3_FIND_VERSION_MINOR}.${Eigen3_FIND_VERSION_PATCH}")
+endif(NOT Eigen3_FIND_VERSION)
+
+macro(_eigen3_check_version)
+  file(READ "${EIGEN3_INCLUDE_DIR}/Eigen/src/Core/util/Macros.h" _eigen3_version_header)
+
+  string(REGEX MATCH "define[ \t]+EIGEN_WORLD_VERSION[ \t]+([0-9]+)" _eigen3_world_version_match "${_eigen3_version_header}")
+  set(EIGEN3_WORLD_VERSION "${CMAKE_MATCH_1}")
+  string(REGEX MATCH "define[ \t]+EIGEN_MAJOR_VERSION[ \t]+([0-9]+)" _eigen3_major_version_match "${_eigen3_version_header}")
+  set(EIGEN3_MAJOR_VERSION "${CMAKE_MATCH_1}")
+  string(REGEX MATCH "define[ \t]+EIGEN_MINOR_VERSION[ \t]+([0-9]+)" _eigen3_minor_version_match "${_eigen3_version_header}")
+  set(EIGEN3_MINOR_VERSION "${CMAKE_MATCH_1}")
+
+  set(EIGEN3_VERSION ${EIGEN3_WORLD_VERSION}.${EIGEN3_MAJOR_VERSION}.${EIGEN3_MINOR_VERSION})
+  if(${EIGEN3_VERSION} VERSION_LESS ${Eigen3_FIND_VERSION})
+    set(EIGEN3_VERSION_OK FALSE)
+  else(${EIGEN3_VERSION} VERSION_LESS ${Eigen3_FIND_VERSION})
+    set(EIGEN3_VERSION_OK TRUE)
+  endif(${EIGEN3_VERSION} VERSION_LESS ${Eigen3_FIND_VERSION})
+
+  if(NOT EIGEN3_VERSION_OK)
+
+    message(STATUS "Eigen3 version ${EIGEN3_VERSION} found in ${EIGEN3_INCLUDE_DIR}, "
+                   "but at least version ${Eigen3_FIND_VERSION} is required")
+  endif(NOT EIGEN3_VERSION_OK)
+endmacro(_eigen3_check_version)
+
+if (EIGEN3_INCLUDE_DIR)
+
+  # in cache already
+  _eigen3_check_version()
+  set(EIGEN3_FOUND ${EIGEN3_VERSION_OK})
+
+else (EIGEN3_INCLUDE_DIR)
+
+  find_path(EIGEN3_INCLUDE_DIR NAMES signature_of_eigen3_matrix_library
+      PATHS
+      ${CMAKE_INSTALL_PREFIX}/include
+      ${KDE4_INCLUDE_DIR}
+      PATH_SUFFIXES eigen3 eigen
+    )
+
+  if(EIGEN3_INCLUDE_DIR)
+    _eigen3_check_version()
+  endif(EIGEN3_INCLUDE_DIR)
+
+  include(FindPackageHandleStandardArgs)
+  find_package_handle_standard_args(Eigen3 DEFAULT_MSG EIGEN3_INCLUDE_DIR EIGEN3_VERSION_OK)
+
+  mark_as_advanced(EIGEN3_INCLUDE_DIR)
+
+endif(EIGEN3_INCLUDE_DIR)
+
Index: kstars/kstars/ksutils.h
===================================================================
--- kstars.orig/kstars/ksutils.h	2013-11-09 17:45:22.758523776 +0100
+++ kstars/kstars/ksutils.h	2013-11-09 17:45:22.754523934 +0100
@@ -29,7 +29,7 @@
 
 #include <cstddef>
 #include <Eigen/Core>
-USING_PART_OF_NAMESPACE_EIGEN
+using namespace Eigen;
 #include <QPointF>
 #include "dms.h"
 
Index: kstars/kstars/projections/projector.h
===================================================================
--- kstars.orig/kstars/projections/projector.h	2013-11-09 17:45:22.758523776 +0100
+++ kstars/kstars/projections/projector.h	2013-11-09 17:45:22.754523934 +0100
@@ -22,7 +22,7 @@
 
 #include <cstddef>
 #include <Eigen/Core>
-USING_PART_OF_NAMESPACE_EIGEN
+using namespace Eigen;
 
 #include <cmath>
 
Index: kstars/kstars/skyglpainter.cpp
===================================================================
--- kstars.orig/kstars/skyglpainter.cpp	2013-11-09 17:45:22.758523776 +0100
+++ kstars/kstars/skyglpainter.cpp	2013-11-09 17:45:22.754523934 +0100
@@ -25,7 +25,7 @@
 #include <cstddef>
 #include <Eigen/Core>
 #include <Eigen/Geometry>
-USING_PART_OF_NAMESPACE_EIGEN
+using namespace Eigen;
 using Eigen::Rotation2Df;
 
 #include <GL/gl.h>
Index: kstars/kstars/skyglpainter.h
===================================================================
--- kstars.orig/kstars/skyglpainter.h	2013-11-09 17:45:22.758523776 +0100
+++ kstars/kstars/skyglpainter.h	2013-11-09 17:45:22.754523934 +0100
@@ -22,7 +22,7 @@
 
 #include <cstddef>
 #include <Eigen/Core>
-USING_PART_OF_NAMESPACE_EIGEN
+using namespace Eigen;
 
 #include "skypainter.h"
 #include "skyobjects/skyobject.h"
