Author: José Manuel Santamaría Lema <panfaust@gmail.com>
Description: Link against pthread
 Linking against libindi requires pthread
Last-Update: 2013-11-20

Index: kstars/kstars/CMakeLists.txt
===================================================================
--- kstars.orig/kstars/CMakeLists.txt	2014-02-07 09:10:50.332325022 +0100
+++ kstars/kstars/CMakeLists.txt	2014-02-07 09:10:50.328324928 +0100
@@ -485,6 +485,7 @@
     ${QT_QTDECLARATIVE_LIBRARY}
     kdeclarative
 	${ZLIB_LIBRARIES}
+        pthread
 	${QT_QTSQL_LIBRARY}
         )
 # qt4_automoc(kstars.h)
